internal/controllers/reference_addon_controller.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	refapisv1alpha1 "github.com/openshift/reference-addon/apis/reference/v1alpha1"
+)
+
+type ReferenceAddonReconciler struct {
+	client.Client
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	latestHeartbeat metav1.Condition
+}
+
+func (r *ReferenceAddonReconciler) Reconcile(
+	ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = r.Log.WithValues("addon", req.NamespacedName.String())
+
+	successfulCondition := metav1.Condition{
+		Type:    "addons.managed.openshift.io/Healthy",
+		Status:  "True",
+		Reason:  "AllComponentsUp",
+		Message: "Everything under reference-addon is working perfectly fine",
+	}
+	failureCondition := metav1.Condition{
+		Type:    "addons.managed.openshift.io/Healthy",
+		Status:  "False",
+		Reason:  "ImproperNaming",
+		Message: "The addon resources are improperly named",
+	}
+
+	// if the ReferenceAddon object getting reconciled has the name "reference-addon", only then report a successful heartbeat
+	if req.NamespacedName.Name == "reference-addon" {
+		r.SetLatestHeartbeat(successfulCondition)
+	} else {
+		r.SetLatestHeartbeat(failureCondition)
+	}
```

Comment:
Can we just report `Success` for now?
The code here would flip/flop the heartbeat condition back and forth when you create two `ReferenceAddon` objects (one called `reference-addon` the other something else...).

Reply:
Shouldn't ideally reference-addon conditionally report both success and failure, just so as to demostrate a more realistic scenario?

Although, I agree that we can tune the logic a bit. Maybe we can go ahead with

```
if req.NamespacedName.Name begins with "redhat-*" or "reference-addon-*"
    report Success
else
    report Failure
```

wdyt :)